src/HttpClient.php

```diff
@@ -289,14 +60,23 @@ public function __construct($disableTls = false, $cafile = null)
      */
     public function download($url)
     {
-        $context = $this->getStreamContext($url);
-        $result = file_get_contents($url, null, $context);
+        if (extension_loaded('zlib')) {
+            array_push($this->headers, "Accept-Encoding: gzip\r\n");
+        }
+
+        if ($this->tls) {
+            humbug_set_headers($this->headers);
+            $result = humbug_get_contents($url);
+        } else {
+            $context = $this->getNonSslStreamContext($url);
+            $result = file_get_contents($url, null, $context);
```

Comment:
We were talking about that with @tgalopin yesterday. If I remove that block we'll be unable to handle `--disable-tls` option. In that case `file_get_contents` has to have it's own context with proxy stuff configured on our side. Does it make sense?